src/interrupts/pins_config.rs

```diff
@@ -0,0 +1,323 @@
+/// Functions related to INT1_AG and INT2_AG interrupt pins configuration
+///
+/// TO DO:
+/// -
+///
+use super::*;
+
+/// Accelerometer/gyroscope interrupt pins common settings
+#[derive(Debug)]
+pub struct PinConfig {
+    // --- CTRL_REG8 REGISTER ---
+    /// Interrupt pin active level (default 0: active high)
+    pub active_level: IntActive,
+    /// Interrupt pin push-pull or open-drain configuration (default 0: push-pull)
+    pub pin_mode: IntPin,
+}
+
+impl Default for PinConfig {
+    fn default() -> Self {
+        PinConfig {
+            active_level: IntActive::High,
+            pin_mode: IntPin::PushPull,
+        }
+    }
+}
+
+impl PinConfig {
+    /// Returns values to be written to CTRL_REG8 register
+    fn ctrl_reg8(&self) -> u8 {
+        let mut data: u8 = 0;
+
+        data |= self.active_level.value() << 5;
+        data |= self.pin_mode.value() << 4;
+
+        data
+    }
+}
+
+/// Accelerometer/gyroscope interrupt pin (INT1_A/G) settings
+#[derive(Debug)]
+pub struct IntConfigAG1 {
+    // --- INT1_CTRL REGISTER ---
+    /// Enable gyroscope interrupt generation on pin INT1_A/G
+    pub enable_gyro_int: Flag,
+    /// Enable accelerometer interrupt generation on pin INT1_A/G
+    pub enable_accel_int: Flag,
+    /// Enable FSS5 interrupt on on pin INT1_A/G
+    pub enable_fss5: Flag,
+    /// Enable overrun interrupt on on pin INT1_A/G
+    pub enable_overrun: Flag,
+    /// Enable FIFO threshold interrupt on on pin INT1_A/G
+    pub enable_fth: Flag,
+    /// Enable boot status interrupt on on pin INT1_A/G
+    pub enable_boot_status: Flag,
+    /// Enable gyroscope data ready interrupt on on pin INT1_A/G
+    pub enable_gyro_dataready: Flag,
+    /// Enable accelerometer data ready interrupt on on pin INT1_A/G
+    pub enable_accel_dataready: Flag,
+}
+
+impl Default for IntConfigAG1 {
+    fn default() -> Self {
+        IntConfigAG1 {
+            enable_gyro_int: Flag::Disabled,
+            enable_accel_int: Flag::Disabled,
+            enable_fss5: Flag::Disabled,
+            enable_overrun: Flag::Disabled,
+            enable_fth: Flag::Disabled,
+            enable_boot_status: Flag::Disabled,
+            enable_gyro_dataready: Flag::Disabled,
+            enable_accel_dataready: Flag::Disabled,
+        }
+    }
+}
+
+impl IntConfigAG1 {
+    /// Returns values to be written to INT1_CTRL register
+    fn int1_ctrl(&self) -> u8 {
```

Comment:
Should not this be a public method?

Reply:
Changed to pub(crate)

Reply:
I think Rust Analyzer still picks it up. But it hides from the doc. Is that your intention?

Reply:
Ideally in my vision the crate user wouldn't know about those internal functions at all, but yes, removing them from the doc is good. So that could work.